### DISCUSSION TITLE
Typescript error on writeContract value

### DISCUSSION THREAD

**Discussion by c0rdeiro** [wevm/wagmi#3844] [Q&A]
wagmi version: 2.5.20
viem version: 2.9.9

Hey, I came across this when integrating the declaration of the wagmi config  [docs](https://wagmi.sh/react/typescript#config-types) to get typesafety.

On `useWriteContract`  I have something like so
```ts
const { writeContract, data } = useWriteContract()
...
writeContract(
      {
        address: 0x..,
        abi: *abi as const*,
        functionName: 'myFunction',
        args: [...someargs],
        value: cost,
      })
``` 
where `cost` is a value that comes from a contract read, so it as a type of `bigint | undefined` and I get the following error: 
`Type 'bigint' is not assignable to type 'undefined'.ts(2322)`

I also tried with `value: 0n` but the problem remains, although I see that by the [docs](https://wagmi.sh/core/api/actions/writeContract#value) it shouldn't give me any error.

Because the code runs and its correct, it's something related to the types declaration I assume.

If anyone could enlighten me, tysm

**Comment by tmm** [comment]
Please attach a minimal reproduction using a [TypeScript Playground](https://www.typescriptlang.org/play).

**Reply by c0rdeiro** [reply]
weirdly enough I was not able to reproduce this, so I'll have to check deep down what's the issue there
